### DISCUSSION TITLE
TLM Dictionary accessible by Fprime Component

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#1773] [Q&A]
Hello,

Are the telemetry id's for all components in a deployment stored in a dictionary accessible by another fprime component? I know that the build step spits out a dictionary the GDS can use, but I am looking for a dictionary more easily accessible from a component itself(i.e a dictionary in an fpp/hpp file). 

The current approach is to manually create this dictionary as an enum in a .hpp file and then include that in my component. However, managing this enum will get messy as the deployment grows in size. 

Context:
I am working on a fault management system and need a component that can monitor specific telemetry channels and see if they are out of bounds. The plan is to use the `TlmGet` input port on `TlmChan` and get the specific IDs I want, however, I need a TLM dictionary to know what the IDs are for each channel. 

Does anyone have any thoughts and/or suggestions?

**Comment by timcanham** [comment]
If specific values are needed internally, you can also use PolyDb to share those values among components. We don't have a built-in feature to share the dictionary for decoding.

**Comment by SterlingPeet** [comment]
I think you could create a cmake target that generates the header file from the dictionary, it just needs to depend on the dictionary cmake target.

@LeStarch is a little more familiar with this sort of cmake wizardry than I am, but we are working on a a tool that does this sort of thing for the Arduino OSAL.
